spec/rspec/support/encoded_string_spec.rb

```diff
@@ -1,51 +1,148 @@
+              resulting_string = build_encoded_string(string, no_converter_encoding).to_s
+              expected_string  = EncodedString::REPLACE
+              expect(resulting_string).to be_identical_string(expected_string, no_converter_encoding)
+            end
+          end
+        end
+
+        # see https://github.com/ruby/ruby/blob/34fbf57aaa/transcode.c#L4289
+        # ISO-8859-1 -> UTF-8 -> EUC-JP
+        # "\xa0" NO-BREAK SPACE, which is available in UTF-8 but not in EUC-JP
+        context 'when there is an undefined conversion to the target encoding' do
+          let(:source_encoding) { Encoding.find('ISO-8859-1') }
+          let(:incompatible_encoding) { Encoding.find('EUC-JP') }
+          let(:string) { "\xa0 hi I am not going to work".force_encoding(source_encoding) }
+
+          it 'normally raises an Encoding::UndefinedConversionError' do
+            expect {
+              string.encode(incompatible_encoding)
+            }.to raise_error(Encoding::UndefinedConversionError)
+          end
+
+          it 'replaces all undefines conversions with the REPLACE string' do
+            resulting_string = build_encoded_string(string, incompatible_encoding).to_s
+            replacement = EncodedString::REPLACE
+            expected_string = "#{replacement} hi I am not going to work"
+            expect(resulting_string).to be_identical_string(expected_string)
+          end
+        end
+      end
 
+      let(:ascii_arrow_symbol) { "\xAE" }
       let(:utf_8_euro_symbol) { "\xE2\x82\xAC" }
 
       describe '#<<' do
         context 'with strings that can be converted to the target encoding' do
+          let(:valid_ascii_string) { "abcdé".force_encoding("ASCII-8BIT") }
+          let(:valid_unicode_string) { utf_8_euro_symbol.force_encoding('UTF-8') }
           it 'encodes and appends the string' do
-            valid_ascii_string = "abc".force_encoding("ASCII-8BIT")
-            valid_unicode_string = utf_8_euro_symbol.force_encoding('UTF-8')
 
-            resulting_string = build_encoded_string(valid_unicode_string, target_encoding) << valid_ascii_string
-            expect(resulting_string).to eq "#{utf_8_euro_symbol}abc".force_encoding('UTF-8')
+            resulting_string = build_encoded_string(valid_unicode_string, utf8_encoding) << valid_ascii_string
+            replacement = EncodedString::REPLACE * 2
+            expected_string = "#{utf_8_euro_symbol}abcd#{replacement}".force_encoding('UTF-8')
+            expect(resulting_string).to be_identical_string(expected_string)
+          end
+
+          it 'copes with encoded strings' do
+            source_encoding = Encoding.find('UTF-16LE')
+            accentless = build_encoded_string("Tu avec carte {count} item has\n", source_encoding)
+            accented   = "Tu avec carté {count} itém has\n".encode(source_encoding)
+            resulting_string = accentless << accented
+            if OS.windows?
+              replacement = "\x82\x82"
```

Comment:
Hmm, that's odd, given we pass `?` as the replacement character and are specifying the encoding.  The question is, does this represent a bug or is it correct behavior on windows?  If it's a bug, I don't want to encode that behavior in a spec that makes us think that's how it should work.  I think I'd prefer that we tag this spec with `:failing_on_appveyor` rather than special case windows, and hopefully someone who uses windows as their primary dev environment can take a look later and figure out if it's desirable behavior or not.